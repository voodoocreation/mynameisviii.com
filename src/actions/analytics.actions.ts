import actionCreatorFactory from "typescript-fsa";

const createAction = actionCreatorFactory("ANALYTICS");

export const trackEvent = createAction<{
  event: string;
  [index: string]: any;
}>("TRACK_EVENT");
