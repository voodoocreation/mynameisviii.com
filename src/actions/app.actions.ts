import actionCreatorFactory from "typescript-fsa";

const createAction = actionCreatorFactory("APP");

export const initApp = createAction.async<{ locale?: string }, {}>("INIT");

export const toggleNavigation = createAction("TOGGLE_NAVIGATION");

export const setOnlineStatus = createAction<boolean>("SET_ONLINE_STATUS");

export const setHasNewVersion = createAction<boolean>("SET_HAS_NEW_VERSION");

export const setCurrentRoute = createAction<string>("SET_CURRENT_ROUTE");

export const changeRoute = createAction.async<string, {}>("CHANGE_ROUTE");
