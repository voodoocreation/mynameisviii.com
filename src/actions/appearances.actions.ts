import actionCreatorFactory from "typescript-fsa";

import { IAppearance, IDynamoResponse, ILatLng } from "../models/root.models";

const createAction = actionCreatorFactory("APPEARANCES");

export const fetchAppearances = createAction.async<
  {},
  IDynamoResponse<IAppearance, "startingAt">
>("FETCH");

export const fetchMoreAppearances = createAction.async<
  {},
  IDynamoResponse<IAppearance, "startingAt">
>("FETCH_MORE");

export const fetchAppearanceBySlug = createAction.async<string, IAppearance>(
  "FETCH_BY_SLUG"
);

export const setCurrentAppearanceSlug = createAction<string>(
  "SET_CURRENT_SLUG"
);

export const geocodeCurrentAppearanceAddress = createAction.async<{}, ILatLng>(
  "GEOCODE_CURRENT_ADDRESS"
);
