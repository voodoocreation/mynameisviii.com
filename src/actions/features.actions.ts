import actionCreatorFactory from "typescript-fsa";

const createAction = actionCreatorFactory("FEATURES");

export const addFeatures = createAction<string | string[]>("ADD");

export const removeFeatures = createAction<string | string[]>("REMOVE");
