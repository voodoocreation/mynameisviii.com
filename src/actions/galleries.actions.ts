import actionCreatorFactory from "typescript-fsa";

import { IGallery, IS3Response } from "../models/root.models";

const createAction = actionCreatorFactory("GALLERIES");

export const fetchGalleries = createAction.async<{}, IS3Response<IGallery>>(
  "FETCH"
);

export const fetchMoreGalleries = createAction.async<{}, IS3Response<IGallery>>(
  "FETCH_MORE"
);

export const fetchGalleryBySlug = createAction.async<string, IGallery>(
  "FETCH_BY_SLUG"
);

export const setCurrentGallerySlug = createAction<string>("SET_CURRENT_SLUG");
