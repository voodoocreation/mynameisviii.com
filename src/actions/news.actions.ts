import actionCreatorFactory from "typescript-fsa";

import { IDynamoResponse, INewsArticle } from "../models/root.models";

const createAction = actionCreatorFactory("NEWS");

export const fetchLatestNews = createAction.async<
  {},
  IDynamoResponse<INewsArticle, "createdAt">
>("FETCH");

export const fetchMoreLatestNews = createAction.async<
  {},
  IDynamoResponse<INewsArticle, "createdAt">
>("FETCH_MORE");

export const fetchNewsArticleBySlug = createAction.async<string, INewsArticle>(
  "FETCH_BY_SLUG"
);

export const setCurrentNewsArticleSlug = createAction<string>(
  "SET_CURRENT_SLUG"
);
