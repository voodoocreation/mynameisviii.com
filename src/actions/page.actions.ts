import actionCreatorFactory from "typescript-fsa";

const actionCreator = actionCreatorFactory("PAGE");

export const toggleNavigation = actionCreator<{}>("TOGGLE_NAVIGATION");

export const changeRoute = actionCreator.async<
  PLChangeRouteStarted,
  PLChangeRouteDone,
  PLChangeRouteFailed
>("CHANGE_ROUTE");
