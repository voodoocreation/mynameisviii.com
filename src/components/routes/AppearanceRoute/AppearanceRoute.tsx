import Head from "next/head";
import * as React from "react";
import { InjectedIntl } from "react-intl";
import { connect } from "react-redux";
import { bindActionCreators, Dispatch } from "redux";
import { ActionCreator } from "typescript-fsa";

import injectIntlIntoPage from "../../../helpers/injectIntlIntoPage";
import { absUrl } from "../../../transformers/transformData";
import Appearance from "../../presentation/Appearance/Appearance";
import ErrorPage from "../../presentation/ErrorPage/ErrorPage";
import Loader from "../../presentation/Loader/Loader";

import * as actions from "../../../actions/root.actions";
import * as selectors from "../../../selectors/root.selectors";

interface IStoreProps {
  appearance?: IAppearance;
  currentLocation?: ILatLng;
  error?: IError;
  isLoading: boolean;
}

interface IDispatchProps {
  fetchAppearanceBySlug: ActionCreator<PLFetchAppearanceBySlugStarted>;
  geocodeCurrentAppearanceAddress: ActionCreator<{}>;
  trackEvent: ActionCreator<PLTrackEvent>;
}

interface IProps extends IStoreProps, IDispatchProps {
  intl: InjectedIntl;
}

class AppearanceRoute extends React.Component<IProps> {
  public static async getInitialProps(props: any) {
    const { query, store } = props.ctx;

    store.dispatch(actions.setCurrentAppearanceSlug(query.slug));

    if (!selectors.getCurrentAppearance(store.getState())) {
      store.dispatch(actions.fetchAppearanceBySlug.started(query.slug));
    }
  }

  public componentWillMount() {
    if (this.props.appearance && !this.props.currentLocation) {
      this.props.geocodeCurrentAppearanceAddress({});
    }
  }

  public render() {
    const { appearance, currentLocation, error, isLoading } = this.props;
    const { formatMessage } = this.props.intl;

    if (isLoading) {
      return <Loader className="PageLoader" />;
    }

    if (!appearance) {
      return <ErrorPage {...error} />;
    }

    return (
      <React.Fragment>
        <Head>
          <title>
            {appearance.title}
            {" Â· "}
            {formatMessage({ id: "BRAND_NAME" })}
          </title>

          <meta content={appearance.description} name="description" />

          <meta property="og:title" content={appearance.title} />
          <meta property="og:description" content={appearance.description} />
          <meta
            property="og:url"
            content={absUrl(`/appearances/${appearance.slug}`)}
          />
          <meta property="og:image" content={appearance.ogImageUrl} />
          <meta property="og:image:width" content="1200" />
          <meta property="og:image:height" content="630" />
        </Head>

        <Appearance
          {...appearance}
          locationLatLng={currentLocation}
          onGalleryInteraction={this.onGalleryInteraction}
        />
      </React.Fragment>
    );
  }

  private onGalleryInteraction = (type: string, index?: number) => {
    this.props.trackEvent({
      event: `appearance.gallery.${type}`,
      index
    });
  };
}

const mapStateToProps = (state: any) => ({
  appearance: selectors.getCurrentAppearance(state),
  currentLocation: selectors.getCurrentAppearanceLocation(state),
  error: selectors.getAppearancesError(state),
  isLoading: selectors.getAppearancesIsLoading(state)
});

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      fetchAppearanceBySlug: actions.fetchAppearanceBySlug.started,
      geocodeCurrentAppearanceAddress:
        actions.geocodeCurrentAppearanceAddress.started,
      trackEvent: actions.trackEvent
    },
    dispatch
  );

export default injectIntlIntoPage(
  connect<IStoreProps, IDispatchProps>(
    mapStateToProps,
    mapDispatchToProps
  )(AppearanceRoute)
);
