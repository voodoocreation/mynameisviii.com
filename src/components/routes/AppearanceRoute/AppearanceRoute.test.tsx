import * as actions from "../../../actions/root.actions";
import { appearance, image } from "../../../models/root.models";
import MockPageContext from "../../../utilities/MockPageContext";
import WrapperWithRedux from "../../../utilities/WrapperWithRedux";
import AppearanceRoute from "./AppearanceRoute";

const item = appearance({
  images: [image({ imageUrl: "1" }), image({ imageUrl: "2" })],
  slug: "test-1"
});

const defaultState = {
  appearances: {
    currentSlug: item.slug,
    items: {
      [item.slug]: item
    }
  }
};

const component = new WrapperWithRedux(AppearanceRoute);

describe("[routes] <AppearanceRoute />", () => {
  describe("getInitialProps", () => {
    const context = new MockPageContext()
      .withDefaultQuery({
        slug: item.slug
      })
      .withDefaultReduxState(defaultState);

    describe("when the appearance isn't already in the store", () => {
      const slug = "new-slug";

      it("calls getInitialProps method", async () => {
        await AppearanceRoute.getInitialProps(
          context.withQuery({ slug }).toObject()
        );
      });

      it("dispatches actions.setCurrentAppearanceSlug with expected payload", () => {
        const matchingActions = context.reduxHistory.filter(
          actions.setCurrentAppearanceSlug.match
        );

        expect(matchingActions).toHaveLength(1);
        expect(matchingActions[0].payload).toBe(slug);
      });

      it("dispatches actions.fetchAppearanceBySlug.started", () => {
        expect(
          context.reduxHistory.filter(
            actions.fetchAppearanceBySlug.started.match
          )
        ).toHaveLength(1);
      });
    });

    describe("when the appearance is already in the store", () => {
      it("calls getInitialProps method", async () => {
        await AppearanceRoute.getInitialProps(context.toObject());
      });

      it("dispatches actions.setCurrentAppearanceSlug", () => {
        expect(
          context.reduxHistory.filter(actions.setCurrentAppearanceSlug.match)
        ).toHaveLength(1);
      });

      it("doesn't dispatch actions.fetchAppearanceBySlug.started", () => {
        expect(
          context.reduxHistory.filter(
            actions.fetchAppearanceBySlug.started.match
          )
        ).toHaveLength(0);
      });
    });
  });

  it("matches snapshot", () => {
    const wrapper = component.withReduxState(defaultState).mount();

    expect(wrapper.render()).toMatchSnapshot();
  });

  it("renders a loader when isLoading is true", () => {
    const wrapper = component
      .withReduxState({
        ...defaultState,
        appearances: { isLoading: true }
      })
      .mount();

    expect(wrapper.find("Loader")).toHaveLength(1);
  });

  it("doesn't render anything when no appearance exists", () => {
    const wrapper = component
      .withReduxState({
        appearances: {
          currentSlug: undefined,
          hasError: true,
          items: {}
        }
      })
      .mount();

    expect(wrapper.render().html()).toBeNull();
  });

  it("tracks gallery interactions correctly", () => {
    const wrapper = component.withReduxState(defaultState).mount();

    wrapper
      .find("ImageGallery Image")
      .last()
      .simulate("click");

    const matchingActions = component.reduxHistory.filter(
      actions.trackEvent.match
    );

    expect(matchingActions).toHaveLength(1);
    expect(matchingActions[0].payload).toEqual({
      event: "appearance.gallery.itemClick",
      index: 1
    });
  });
});
