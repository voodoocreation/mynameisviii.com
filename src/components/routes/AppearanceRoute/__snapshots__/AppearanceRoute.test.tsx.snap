// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[routes] <AppearanceRoute /> matches snapshot 1`] = `
<article
  class="Appearance"
>
  <header
    class="PageHeader"
  >
    <h1 />
  </header>
  <div
    class="MetaBar Appearance--meta"
  >
    <div
      class="Meta Appearance--date"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
        />
      </svg>
       
      <span
        class="Meta--label"
      >
        Date:
      </span>
       
      <span
        class="Meta--value"
      >
        <time
          class="DateTime"
          datetime="2018-01-01"
          title="Monday, 1 January 2018"
        >
          Monday, 1 January 2018
        </time>
      </span>
    </div>
    <div
      class="Meta Appearance--time"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z"
        />
      </svg>
       
      <span
        class="Meta--label"
      >
        Time:
      </span>
       
      <span
        class="Meta--value"
      >
        <time
          class="DateTime"
          datetime="2018-01-01T00:00:00"
          title="12:00 AM"
        >
          12:00 AM
        </time>
        â€“
        <time
          class="DateTime"
          datetime="2018-01-01T00:00:00"
          title="12:00 AM"
        >
          12:00 AM
        </time>
      </span>
    </div>
    <div
      class="Meta Appearance--location"
      title=" - undefined"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
        />
      </svg>
       
      <span
        class="Meta--label"
      >
        Location:
      </span>
       
      <span
        class="Meta--value"
      />
    </div>
  </div>
  <div
    class="Appearance--body"
  >
    <div
      class="Appearance--details"
    >
      <section
        class="Appearance--description"
      >
        <p />
      </section>
      <section
        class="Appearance--acts"
      >
        <h2>
          Featured acts
        </h2>
        <div
          class="Appearance--acts--items"
        />
      </section>
    </div>
    <section
      class="Appearance--images"
    >
      <h2>
        Photos
      </h2>
      <div
        class="ImageGallery Appearance--images--items"
      >
        <figure
          class="Image isLoading"
          role="button"
          tabindex="0"
        >
          <div
            class="Image--wrapper"
          >
            <img
              alt=""
              src="1"
            />
            <span
              class="Loader"
            >
              <div
                class="Loader--animation"
              >
                <span />
                <span />
                <span />
              </div>
            </span>
          </div>
        </figure>
        <figure
          class="Image isLoading"
          role="button"
          tabindex="0"
        >
          <div
            class="Image--wrapper"
          >
            <img
              alt=""
              src="2"
            />
            <span
              class="Loader"
            >
              <div
                class="Loader--animation"
              >
                <span />
                <span />
                <span />
              </div>
            </span>
          </div>
        </figure>
      </div>
    </section>
  </div>
</article>
`;
